WuJinhan1 iP

WuJinhan1 opened the pull request:
# Jard
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Jard frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use
+ All you need to do is,

download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 🧌
4. And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

WuJinhan1 committed:
Level 0 completed

WuJinhan1 committed:
finished level 1

WuJinhan1 committed:
finished level 2

WuJinhan1 committed:
finished level 3

WuJinhan1 committed:
finished level 4

WuJinhan1 committed:
finished level 5

WuJinhan1 committed:
Automated Text UI Testing

WuJinhan1 committed:
finished level 6

WuJinhan1 committed:
made comments better

WuJinhan1 committed:
Add Level-7

WuJinhan1 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Level-7

WuJinhan1 committed:
Add Level 8

WuJinhan1 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add Level 8

WuJinhan1 committed:
Add A-MoreOOP

WuJinhan1 committed:
Revert "Add A-MoreOOP"

This reverts commit f2831a0153fed7a0b372eacb3c1f33c2ecaa5550.

WuJinhan1 committed:
Add A-MoreOOP

WuJinhan1 committed:
Add A-Packages

WuJinhan1 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

WuJinhan1 committed:
Add gradle

WuJinhan1 committed:
Add A-JUnit

WuJinhan1 committed:
Add A-JavaDoc

WuJinhan1 committed:
Add A-CodingStandard

WuJinhan1 committed:
Add Level-9

WuJinhan1 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Level-9

WuJinhan1 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add A-CodingStandard

WuJinhan1 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add A-JavaDoc

# Conflicts:
#	src/main/java/Jard/Jard.java
#	src/main/java/Jard/Ui.java

jevan13 commented on src/main/java/Jard/Jard.java:
```diff
@@ -0,0 +1,101 @@
+            String[] inputParts = Parser.parseCommand(input);
+            String command = inputParts[0];
+
+            try {
```
Should all these operations be done in a separate class, eg. TaskList class?

jevan13 commented on src/main/java/Jard/Parser.java:
```diff
@@ -0,0 +1,104 @@
+ * The Parser class is responsible for parsing user input commands, tasks, and dates.
+ * It converts user input strings into corresponding task objects or other necessary data types.
+ */
+public class Parser {
```
LGTM!

jevan13 commented on src/main/java/Jard/Storage.java:
```diff
@@ -0,0 +1,117 @@
+ * The Storage class handles the reading and writing of task data to and from a file.
+ * It is responsible for loading tasks from a file upon startup and saving tasks to a file when changes are made.
+ */
+public class Storage {
```
LGTM!

jevan13 commented on src/main/java/Jard/Task.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
```
should markAsDone() and markAsNotDone() can be combined as one method that takes in a boolean?

jevan13 commented on src/main/java/Jard/Ui.java:
```diff
@@ -0,0 +1,155 @@
+/**
+ * Handles user interactions and displays messages to the user.
+ */
+public class Ui {
```
LGTM!

jevan13 commented on src/main/java/Jard/Task.java:
```diff
@@ -0,0 +1,72 @@
+     *
+     * @return The description of the task.
+     */
+    public String getDescription() {
```
Is there a need to repeat this for all child classes?

jevan13 reviewed (commented):
Looks good.
Some suggestions for your consideration.

thony-ui commented on src/main/java/Jard/Jard.java:
You can consider making the operations done in a separate class, for example ```class execute```

thony-ui commented on src/main/java/Jard/Parser.java:
Avoid having nested if statements to make the code more readable

thony-ui commented on src/main/java/Jard/Task.java:
Good use of abstract class

thony-ui commented on src/test/java/Jard/DeadlineTest.java:
Should test for edge cases. Whats the expected output if its the wrong output etc

WuJinhan1 committed:
finished level-10

WuJinhan1 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  finished level-10

WuJinhan1 committed:
Add assert statements to document key assumptions in the code

Using Java's assert feature to document important assumptions that should hold at various points in the code helps improve code reliability.

Added assert statements like in the getJardResponse method to document the assumption that both the text and img parameters should not be null.

WuJinhan1 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Add assert statements to document key assumptions in the code

WuJinhan1 committed:
Add C-BetterSearch

WuJinhan1 committed:
Add User guide

WuJinhan1 committed:
Add A-UserGuide

WuJinhan1 committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add A-UserGuide
  Add User guide

WuJinhan1 committed:
Add A-Release

WuJinhan1 committed:
Add reuse declarations

WuJinhan1 committed:
Add A-Release
